lightning/src/ln/channelmanager.rs

```diff
@@ -3543,12 +3545,13 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 	fn get_htlc_temp_fail_err_and_data(&self, desired_err_code: u16, scid: u64, chan: &Channel<Signer>) -> (u16, Vec<u8>) {
 		debug_assert_eq!(desired_err_code & 0x1000, 0x1000);
 		if let Ok(upd) = self.get_channel_update_for_onion(scid, chan) {
-			let mut enc = VecWriter(Vec::with_capacity(upd.serialized_length() + 4));
+			let mut enc = VecWriter(Vec::with_capacity(upd.serialized_length() + 6));
 			if desired_err_code == 0x1000 | 20 {
 				// TODO: underspecified, follow https://github.com/lightning/bolts/issues/791
 				0u16.write(&mut enc).expect("Writes cannot fail");
 			}
-			(upd.serialized_length() as u16).write(&mut enc).expect("Writes cannot fail");
+			(upd.serialized_length() as u16 + 2).write(&mut enc).expect("Writes cannot fail");
```

Comment:
ad this point, why not add a new method like the following one, It makes the code cleaner and failure resistant to future changes
```suggestion
			(upd.serialized_length_without_prefix() as u16).write(&mut enc).expect("Writes cannot fail");
```

Reply:
> ad this point, why not add a new method like the following one, It makes the code cleaner and failure resistant to future changes

I don't think this is feasible because `serialized_length()` is **not** a method of `ChannelUpdate`, but part of the `Writable` trait. This kind of thing was what I referred to above when I said that the solution feels a bit 'hacky', but does not justify a larger scale refactoring of (de-)serialization.

Reply:
Sorry, I was missed that it was implemented in `serialized_lenth()`, you are right! the refactoring is not justified for a single use case